File: src/main/java/me/zeroeightsix/kami/module/modules/player/PacketCancel.java
package me.zeroeightsix.kami.module.modules.player;

import me.zero.alpine.listener.EventHandler;
import me.zero.alpine.listener.Listener;
import me.zeroeightsix.kami.event.events.PacketEvent;
import me.zeroeightsix.kami.module.Module;
import me.zeroeightsix.kami.setting.Setting;
import me.zeroeightsix.kami.setting.Settings;
import net.minecraft.network.play.client.*;

/**
 * @author dominikaaaa
 */
@Module.Info(
        name = "PacketCancel",
        description = "Cancels specific packets used for various actions",
        category = Module.Category.PLAYER
)
public class PacketCancel extends Module {
    private Setting<Boolean> all = register(Settings.b("All", false));
    private Setting<Boolean> packetInput = register(Settings.booleanBuilder("CPacketInput").withValue(true).withVisibility(v -> !all.getValue()));
    private Setting<Boolean> packetPlayer = register(Settings.booleanBuilder("CPacketPlayer").withValue(true).withVisibility(v -> !all.getValue()));
    private Setting<Boolean> packetEntityAction = register(Settings.booleanBuilder("CPacketEntityAction").withValue(true).withVisibility(v -> !all.getValue()));
    private Setting<Boolean> packetUseEntity = register(Settings.booleanBuilder("CPacketUseEntity").withValue(true).withVisibility(v -> !all.getValue()));
    private Setting<Boolean> packetVehicleMove = register(Settings.booleanBuilder("CPacketVehicleMove").withValue(true).withVisibility(v -> !all.getValue()));
    private int numPackets;
    
    @EventHandler
    private final Listener<PacketEvent.Send> sendListener = new Listener<>(event -> {
        if (
            (all.getValue())
            ||
            (packetInput.getValue() && event.getPacket() instanceof CPacketInput)
            ||
            (packetPlayer.getValue() && event.getPacket() instanceof CPacketPlayer)
            ||
            (packetEntityAction.getValue() && event.getPacket() instanceof CPacketEntityAction)
            ||
            (packetUseEntity.getValue() && event.getPacket() instanceof CPacketUseEntity)
            ||
            (packetVehicleMove.getValue() && event.getPacket() instanceof CPacketVehicleMove)
        ) {
            event.cancel();
            numPackets++;
        }
    });

    public void onDisable() {
        numPackets = 0;
    }

    @Override
    public String getHudInfo() {
        return Integer.toString(numPackets);
    }
}

File: src/main/java/me/zeroeightsix/kami/module/modules/player/PacketCancel.kt
package me.zeroeightsix.kami.module.modules.player

import me.zeroeightsix.kami.event.events.PacketEvent
import me.zeroeightsix.kami.module.Category
import me.zeroeightsix.kami.module.Module
import net.minecraft.network.play.client.*
import org.kamiblue.event.listener.listener

internal object PacketCancel : Module(
    name = "PacketCancel",
    description = "Cancels specific packets used for various actions",
    category = Category.PLAYER
) {
    private val all by setting("All", false)
    private val packetInput by setting("CPacketInput", true, { !all })
    private val packetPlayer by setting("CPacketPlayer", true, { !all })
    private val packetEntityAction by setting("CPacketEntityAction", true, { !all })
    private val packetUseEntity by setting("CPacketUseEntity", true, { !all })
    private val packetVehicleMove by setting("CPacketVehicleMove", true, { !all })

    private var numPackets = 0

    override fun getHudInfo(): String {
        return numPackets.toString()
    }

    init {
        listener<PacketEvent.Send> {
            if (all
                || it.packet is CPacketInput && packetInput
                || it.packet is CPacketPlayer && packetPlayer
                || it.packet is CPacketEntityAction && packetEntityAction
                || it.packet is CPacketUseEntity && packetUseEntity
                || it.packet is CPacketVehicleMove && packetVehicleMove
            ) {
                it.cancel()
                numPackets++
            }
        }

        onDisable {
            numPackets = 0
        }
    }
}